awx/main/models/projects.py

```diff
@@ -455,11 +464,12 @@ def _get_related_jobs(self):
         )
 
     def delete(self, *args, **kwargs):
-        path_to_delete = self.get_project_path(check_if_exists=False)
+        paths_to_delete = (self.get_project_path(check_if_exists=False), self.get_cache_path())
         r = super(Project, self).delete(*args, **kwargs)
-        if self.scm_type and path_to_delete:  # non-manual, concrete path
-            from awx.main.tasks import delete_project_files
-            delete_project_files.delay(path_to_delete)
+        for path_to_delete in paths_to_delete:
+            if self.scm_type and path_to_delete:  # non-manual, concrete path
+                from awx.main.tasks import delete_project_files
+                delete_project_files.delay(path_to_delete)
```

Comment:
This isn't something you need to address, just FYI. Running a task now goes over NOTIFY/LISTEN interface. `NOTIFY` has a maximum payload size of 8,000. Apply that to this case and if the path string is longer than 8,000 characters, the cleanup will not run.

Here is a pattern when the payload is large.

https://github.com/ansible/awx/blob/d8f6ea3fe89eaa6aea375dfa0280b43c66899ae0/awx/api/generics.py#L987
https://github.com/ansible/awx/blob/d8f6ea3fe89eaa6aea375dfa0280b43c66899ae0/awx/main/tasks.py#L2938

Reply:
huh, I would think that could be a problem for the `handle_setting_changes` task, but it looks like the 8,000 limit may be specifically selected to be more than the character count for a listing of all setting keys, which comes to 6,845.

The project folder _could_ be anywhere if the user configured it in CTiT, but that's not a use case that concerns me a lot. The project name is in both of these paths, but that's capped at 512. It actually helps that we issue the task twice, instead of putting more data in the args.